packages/slice-machine/src/routerProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from "react";
+
+import { useSelector } from "react-redux";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { userHasDoneTheOnboarding } from "@src/modules/userContext";
+import { useRouter } from "next/router";
+
+const RouterProvider: React.FunctionComponent = ({ children }) => {
```

Comment:
I've seen how you use this Provider and I think it's wrong.
I would prefer to have it not as a provider but as a custom hook.
The only thing that differentiate it from a custom hook here is that you return the children as it is, you don't modify them.

Reply:
Fixed :)